src/ThemeToggle.svelte

```diff
@@ -0,0 +1,187 @@
+<script lang="ts">
+  import { onMount } from 'svelte';
+
+  let theme: "light" | "dark" | undefined = undefined;
+
+  onMount(() => {
+    let storedTheme = window.localStorage.getItem('theme');
+
+    if (storedTheme === null) {
+      // If no theme stored, set it to system default.
+      if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
+        theme = "dark";
+      } else {
+        theme = "light";
+      }
+    } else if (storedTheme === "light" || storedTheme === "dark") {
+      theme = storedTheme;
+    }
+  });
+
+  $: if (theme) {
+    document.documentElement.setAttribute("data-theme", theme);
+    window.localStorage.setItem('theme', theme);
+  }
+</script>
+
+<style>
+  .theme-button {
+    user-select: none;
+    cursor: pointer;
+    width: 40px;
+    height: 40px;
+    margin-left: 8px;
+  }
+</style>
+
+<div
+  class="theme-button"
+  role="button"
+  on:click={() => {
+    theme = theme === "dark" ? "light" : "dark";
+  }}
+>
+  {#if theme === "dark"}
+    <svg
```

Comment:
I would propose to move the `dark` and `light` svgs to the `Icon` component, to keep all the svgs there when possible

Reply:
ec07fc3fe2e9b0be0f279f65eab5798bafd5fca6